be/src/vec/exec/scan/vmeta_scanner.cpp

```diff
@@ -158,38 +153,48 @@ Status VMetaScanner::_fill_block_with_remote_data(const std::vector<MutableColum
     return Status::OK();
 }
 
-Status VMetaScanner::_fetch_iceberg_metadata_batch() {
-    VLOG_CRITICAL << "VMetaScanner::_fetch_iceberg_metadata_batch";
+Status VMetaScanner::_fetch_metadata(const TMetaScanRange& meta_scan_range) {
+    VLOG_CRITICAL << "VMetaScanner::_fetch_iceberg_metadata";
+    switch (meta_scan_range.metadata_type) {
+    case TMetadataType::ICEBERG:
+        return _fetch_iceberg_metadata(meta_scan_range);
+    default:
+        _meta_eos = true;
+        break;
+    }
+    return Status::OK();
+}
+
+Status VMetaScanner::_fetch_iceberg_metadata(const TMetaScanRange& meta_scan_range) {
```

Comment:
the RPC logic is identical for all kinds of metadata fetch request.
I think it should be extracted to a method.
And for different request type, use different method named as `buildXXXRequest()`

Reply:
done